src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1052,6 +1052,9 @@ def join(
         right: "OrderedDataFrame",
         left_on_cols: Optional[list[str]] = None,
         right_on_cols: Optional[list[str]] = None,
+        left_match_condition: Optional[str] = None,
```

Comment:
this seems for column identifier, let's be consistent on the name, and call it left_match_col, and right_match_col

Reply:
so there is only one match column is allowed for Pandas? what is the error we will see if more than on match column is provided?

Reply:
Good catch, added some error checking in the frontend with negative tests to match.